gluten-core/src/main/scala/org/apache/gluten/expression/ExpressionMappings.scala

```diff
@@ -354,6 +354,16 @@ object ExpressionMappings {
       .toMap[Class[_], String]
   }
 
-  var expressionExtensionTransformer: ExpressionExtensionTrait =
-    DefaultExpressionExtensionTransformer()
+  private var expressionExtensionTransformer: Option[ExpressionExtensionTrait] = None
+
+  def getExpressionExtensionTransformer: ExpressionExtensionTrait = {
+    expressionExtensionTransformer.getOrElse(new DefaultExpressionExtensionTransformer)
```

Comment:
Can we throw when it's none? Otherwise we don't make it sure whether the change (setting it in executor side) takes effect. Say if we throw when none, then we must set it in both driver and executor sides otherwise the application will fail by error.

Reply:
@zhztheplayer Yes. Updated.